### ISSUE TITLE
found the "/var/run/httpd.pid" but exiting script

### ISSUE THREAD

**Issue by Anasxrt** [richardforth/apache2buddy#222]
I run the script on my OpenVZ VPS. I found the result as fallowing which the script found "/var/run/httpd.pid" but still exit. Please advise

#########################################################
apache2buddy.pl report for vps.abc.com (103.xxx.xxx.xxx)
#########################################################
[ OK ] This script is being run as root.
[ OK ] The utility 'pmap' exists and is available for use: /usr/bin/pmap
[ OK ] The utility 'netstat' exists and is available for use: /bin/netstat
[ OK ] 'php' exists and is available for use: /usr/local/bin/php
[ OK ] The utility 'apachectl' exists and is available for use: /usr/sbin/apachectl
[ OK ] The 'python' binary exists and is available for use: /usr/bin/python
[ OK ] The port (port 80) is a valid port.
[ -- ] We are attempting to discover the operating system type and version number ...
[ -- ] Distro: CentOS
[ -- ] Version: 6.9
[ -- ] Codename: Final
[ OK ] This distro is supported by apache2buddy.pl.
[ -- ] Hostname: vps.fahmui.com
[ -- ] Primary IP: 103.xxx.xxx.xxx
[ -- ] We are checking the service running on port 80...
[ -- ] The process listening on port 80 is /usr/sbin/varnishd
[ -- ] The process running on port 80 is not Apache. Falling back to process list...
[ -- ] Apache is actually listening on port 8080
[ -- ] The process running on port 8080 is Apache/2.4.27 (Unix).
[ -- ] The full path to the Apache config file is: /etc/httpd/conf/httpd.conf
[ -- ] Apache is using prefork model.
[ -- ] pidfile setting is "/var/run/httpd.pid".
[ !! ] Unable to locate pid file. Exiting.	  

####Checking PID ####
[root@vps ~]# ll /var/run/httpd.pid 
-rw-r--r-- 1 root root 5 Oct  8 00:11 /var/run/httpd.pid
[root@vps ~]#

**Comment by richardforth** [comment]
Ack.

Sorry been busy, will look at this hopefully tonight  - or tomorrow evening

**Comment by richardforth** [comment]
Looking at codebase, this snippet should have found it:

```perl
		if ( -f $pidfile_cfv ) {
			our $pidfile =$pidfile_cfv;
		}
```

After this point we just start guessing.

Im not sure why its not detecting the pidfile as a real file.

**Comment by richardforth** [comment]
I was able to replicate this on centos7 / apache2.4 by manually editting the pidfile directive:

VERBOSE: Searching Apache configuration for the pidfile directive
VERBOSE: "/var/run/httpd.pid"
[ -- ] pidfile setting is "/var/run/httpd.pid".
VERBOSE: Looking for pid file ...[ !! ] Unable to locate pid file. Exiting.


No idea why this is failing but at least i can replicate now which means we are closer to a solution! :)

**Comment by richardforth** [comment]
[root@perl ~]# ls -la /var/run/httpd.pid
-rw-r--r--. 1 root root 5 Oct 13 15:59 /var/run/httpd.pid

**Comment by richardforth** [comment]
ok sorted

the configuration file is broken

This fails:

PidFile "/var/run/httpd.pid"

This Works

PidFile /var/run/httpd.pid

**Comment by richardforth** [comment]
If you want I can write a work around, by stripping any quotes from the returned string

"/var/run/httpd.pid" -> /var/run/httpd.pid

**Comment by richardforth** [comment]
http://www.perlmonks.org/bare/?node_id=85859

**Comment by richardforth** [comment]
VERBOSE: "/var/run/httpd.pid"
[ -- ] pidfile setting is "/var/run/httpd.pid".
VERBOSE: Stripping any quotes from string ...
VERBOSE: BEFORE ("/var/run/httpd.pid").
VERBOSE: AFTER (/var/run/httpd.pid).
[ -- ] Actual pidfile is /var/run/httpd.pid.

**Comment by richardforth** [comment]
```perl
#  Fix for Issue #222 strip any quotes from reurned string
                #  "/var/run/httpd.pid" becomes /var/run/httpd.pid
                if ($VERBOSE) { print "VERBOSE: Stripping any quotes from string ...\n" }
                if ($VERBOSE) { print "VERBOSE: BEFORE ($pidfile_cfv).\n" }
                $pidfile_cfv =~ s/^"(.*)"$/$1/;
                $pidfile_cfv =~ s/^'(.*)'$/$1/;
                if ($VERBOSE) { print "VERBOSE: AFTER ($pidfile_cfv).\n" }
                if ( -f $pidfile_cfv ) {
                        our $pidfile =$pidfile_cfv;
                }
```
